### ISSUE TITLE
afk plugin doesn't work

### ISSUE THREAD

**Issue by campbellr** [StarryPy/StarryPy#157]
I noticed that when i set myself AFK with the `/afk` command, i immediately get my AFK state unset.

I looked at the code and it turns out that this is a general problem with all `SimpleCommandPlugin` subclasses that use the `on_chat_sent` callback.

What happens is that in `PluginManager.do` we iterate over the list of plugins (which is in arbitrary order due to the fact that it is stored in a dictionary). In my case, `CommandPlugin.on_chat_sent` callback gets called, which triggers `AFKCommand.afk`, which sets my user AFK.

Unfortunately, then AFKCommand.on_chat_send` callback gets called after that for the same `/afk` chat message, and unsets my AFK state.

I ended up with the following fix, although it seems pretty hacky:

```diff
diff --git a/starrypy/plugin_manager.py b/starrypy/plugin_manager.py
index 1a102dd..2f4c78a 100644
--- a/starrypy/plugin_manager.py
+++ b/starrypy/plugin_manager.py
@@ -201,7 +201,13 @@ class PluginManager(object):
         return_values = []
         if protocol is None:
             return True
-        for plugin in self.plugins.itervalues():
+        
+        # Ensure CommandPlugin is always called last, so that it doesn't
+        # interfere with SimpleCommandPlugin subclasses
+        def key(item):
+            return 1 if item.name == 'command_plugin' else 0
+
+        for plugin in sorted(self.plugins.itervalues(), key=key):
             try:
                 if not plugin.active:
                     continue

```

If no one has a better solution, I'll submit a pull request.

One concern i have with this is that this code is on the critical path, so constantly sorting the list of plugins could have a performance impact (although not one i've noticed). An alternative would be store the plugins in a list instead of a dictionary, and pre-sort the list.

**Comment by 8r2y5** [comment]
Using itervalues (which is generator) with sorted (that need all elements) is pointless. It will be much better to put sorted plugins into `OrderedDict` when loading plugins. Then ensure that order is correct on activating/deactivating plugins.

Please before you do any thing review this PR: https://github.com/StarryPy/StarryPy/pull/148

**Comment by campbellr** [comment]
You're right, OrderedDict (or simply a list, i don't see a anything that requires it to be a dictionary) would be better. I alluded to that in my last comment, but didn't want to make any major changes without input.

I'll take a look at #148.
